### PR TITLE
Chartbeat author url bug

### LINKED ISSUE #3040: Chartbeat author/URL bugs
1) On canonical, the `g1` value is set to `No Author`, whereas on AMP this value is left empty.  I'm not particularly fussed which we go with but we should be constistent.

2) On canonical only, when using the inline link to move between articles (so moving from one page to another within the SPA), the `p` value does not update to reflect the URL of the second article

_Originally posted by @jimjohnsonrollings in https://github.com/bbc/simorgh/issues/2808#issuecomment-519854791_

### PR DESCRIPTION
**Pull request by rhenshaw56**
Resolves #3040 

**Overall change:** Fix bug with updating chartbeat config when clicking inline link, added author value to config

**Code changes:**

- Added author value to config
- Added current path to config
- Added current path to request context
- Update snapshots

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/lib/analyticsUtils/chartbeat/index.js [hunk 1]
```diff
@@ -65,3 +65,9 @@ export const getTitle = (pageType, pageData, brandName) => {
       return null;
   }
 };
+
+export const getAuthors = (pageType, authors) => {
```
Comment (RayNjeri) [inline src/app/lib/analyticsUtils/chartbeat/index.js]: Consider refactoring 
`export const getAuthors = (pageType, authors) =>  (pageType === 'article' && authors )? authors : 'No Author';`

**Comment by 12** [comment]
This is sending two network requests to `ping.chartbeat.net` on page load that have different query params/values
```
http://ping.chartbeat.net/ping?h=test.bbc.co.uk&p=%2Fnews%2Farticles%2Fc0g992jmmkko&u=CrAdAKtKLWsCLGNcG&d=localhost.bbc.com&g=50924&g0=News%2C%20News%20-%20ART&g1=https%3A%2F%2Fwww.facebook.com%2Fbbcnews&g4=New%20Article&n=0&f=e6345&c=0&x=0&m=0&y=3828&o=1093&w=1066&j=45&R=1&W=0&I=0&E=0&e=0&r=&b=277&t=D0wMo7DlEhOvDQ4wrvCasCfPC8QYSb&V=116&z=t%3DDqF_FJBtTwNlBtY6nsY7js8B8i3X4%26E%3D4%26x%3D0%26c%3D0.08%26y%3D3850%26w%3D1066&i=Royal%20wedding%20flowers%20given%20to%20Hackney%20hospice&tz=-60&sn=1&sv=BM6Su5HHgLwDFGkmQC5VRrAD277Bt&sd=7&im=040b0713&_
```
and
```
http://ping.chartbeat.net/ping?h=test.bbc.co.uk&p=%2Fnews%2Farticles%2Fc0g992jmmkko&u=CrAdAKtKLWsCLGNcG&d=localhost.bbc.com&g=50924&g0=News%2C%20News%20-%20ART&g1=https%3A%2F%2Fwww.facebook.com%2Fbbcnews&g4=New%20Article&n=0&f=e6345&c=0&x=0&m=0&y=3850&o=1093&w=1066&j=45&R=1&W=0&I=0&E=0&e=0&v=http%3A%2F%2Flocalhost.bbc.com%2Fnews%2Farticles%2Fc0g992jmmkko&vp=1&b=1078&t=biZqzDGUs8rBrXq420ybq0T_C8J&V=116&z=t%3DD0wMo7DlEhOvDQ4wrvCasCfPC8QYSb%26E%3D0%26x%3D0%26c%3D0.01%26y%3D3850%26w%3D1066&i=Royal%20wedding%20flowers%20given%20to%20Hackney%20hospice&tz=-60&sn=1&sv=BM6Su5HHgLwDFGkmQC5VRrAD277Bt&sd=8&im=04810713&_
```

**Comment by jamesdonoh** [comment]
Rhys' observation definitely needs investigating.

Also we need to make sure this is tested as working as expected for front pages too as Chartbeat has been added recently for them too.

File: src/app/containers/ArticleMain/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -3,6 +3,7 @@
 exports[`ArticleMain should render a news article correctly 1`] = `
 <RequestContextProvider
   bbcOrigin="https://www.test.bbc.co.uk"
+  currentPath={null}
```
Comment (12) [inline src/app/containers/ArticleMain/__snapshots__/index.test.jsx.snap]: Should this really be `null`? I'd expect `currentPath` to always have a value...

File: src/app/containers/PageHandlers/__snapshots__/withContexts.test.jsx.snap [hunk 3]
```diff
@@ -14,6 +14,7 @@ exports[`withContexts HOC should return all context providers 1`] = `
     >
       <RequestContextProvider
         bbcOrigin="https://www.bbc.com"
+        currentPath={null}
```
Comment (12) [inline src/app/containers/PageHandlers/__snapshots__/withContexts.test.jsx.snap]: Same question as before

File: src/app/containers/PageHandlers/withContexts.jsx [hunk 4]
```diff
@@ -46,12 +48,14 @@ const WithContexts = Component => {
     previousPath: string,
     service: string.isRequired,
     dials: objectOf(bool).isRequired,
+    currentPath: string,
```
Comment (12) [inline src/app/containers/PageHandlers/withContexts.jsx]: We should always have a `currentPath` right? I don't see why this can't be a `required` prop.

**Review by dr3 — DISMISSED** [review]
Can we not just use location.pathname wherever we use it?

**Comment by rhenshaw56** [comment]
> This is sending two network requests to `ping.chartbeat.net` on page load that have different query params/values
> 
> ```
> http://ping.chartbeat.net/ping?h=test.bbc.co.uk&p=%2Fnews%2Farticles%2Fc0g992jmmkko&u=CrAdAKtKLWsCLGNcG&d=localhost.bbc.com&g=50924&g0=News%2C%20News%20-%20ART&g1=https%3A%2F%2Fwww.facebook.com%2Fbbcnews&g4=New%20Article&n=0&f=e6345&c=0&x=0&m=0&y=3828&o=1093&w=1066&j=45&R=1&W=0&I=0&E=0&e=0&r=&b=277&t=D0wMo7DlEhOvDQ4wrvCasCfPC8QYSb&V=116&z=t%3DDqF_FJBtTwNlBtY6nsY7js8B8i3X4%26E%3D4%26x%3D0%26c%3D0.08%26y%3D3850%26w%3D1066&i=Royal%20wedding%20flowers%20given%20to%20Hackney%20hospice&tz=-60&sn=1&sv=BM6Su5HHgLwDFGkmQC5VRrAD277Bt&sd=7&im=040b0713&_
> ```
> 
> and
> 
> ```
> http://ping.chartbeat.net/ping?h=test.bbc.co.uk&p=%2Fnews%2Farticles%2Fc0g992jmmkko&u=CrAdAKtKLWsCLGNcG&d=localhost.bbc.com&g=50924&g0=News%2C%20News%20-%20ART&g1=https%3A%2F%2Fwww.facebook.com%2Fbbcnews&g4=New%20Article&n=0&f=e6345&c=0&x=0&m=0&y=3850&o=1093&w=1066&j=45&R=1&W=0&I=0&E=0&e=0&v=http%3A%2F%2Flocalhost.bbc.com%2Fnews%2Farticles%2Fc0g992jmmkko&vp=1&b=1078&t=biZqzDGUs8rBrXq420ybq0T_C8J&V=116&z=t%3DD0wMo7DlEhOvDQ4wrvCasCfPC8QYSb%26E%3D0%26x%3D0%26c%3D0.01%26y%3D3850%26w%3D1066&i=Royal%20wedding%20flowers%20given%20to%20Hackney%20hospice&tz=-60&sn=1&sv=BM6Su5HHgLwDFGkmQC5VRrAD277Bt&sd=8&im=04810713&_
> ```

@12 I reached out to Chartbeat support on this and they confirmed that this is an expected behavior, which also the same behaviour on http://www.bbc.com/arabic

**Review by RayNjeri — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by 12 — CHANGES_REQUESTED** [review]
We need to remove the `author` tag from the Chartbeat config

Other than that, looks good

**Comment by rhenshaw56** [comment]
> We need to remove the `author` tag from the Chartbeat config
> 
> Other than that, looks good

@12 done!

**Review by 12 — APPROVED** [review]
Nice one 👍

**Pull request merged**